<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        
        <meta name="author" content="Torsten Seemann, Anna Syme">
        
        <link rel="shortcut icon" href="../../../favicon.ico">
        

	<title>Comparative genomics - Sepsis Tutorials</title>

        <link href="../../../css/bootstrap-custom.min.css" rel="stylesheet">
        <link href="../../../css/font-awesome-4.0.3.css" rel="stylesheet">
        <link rel="stylesheet" href="../../../css/highlight.css">
        <link href="../../../css/base.css" rel="stylesheet">
        <link href="../../../css/sepsis.css" rel="stylesheet">

        <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
        <!--[if lt IE 9]>
            <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
            <script src="https://oss.maxcdn.com/libs/respond.js/1.3.0/respond.min.js"></script>
        <![endif]-->

        
    </head>

    <body>

        <div class="navbar navbar-default navbar-fixed-top" role="navigation">
    <div class="container">

        <!-- Collapsed navigation -->
        <div class="navbar-header">
            <!-- Expander button -->
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>

            <!-- Main title -->
            <a class="navbar-brand" href="../../..">Sepsis Tutorials</a>
        </div>

        <!-- Expanded navigation -->
        <div class="navbar-collapse collapse">
            <!-- Main navigation -->
            <ul class="nav navbar-nav">
            
            
                <li >
                    <a href="../../..">Home</a>
                </li>
            
            
            
                <li class="dropdown active">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Workshops <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        
  <li class="dropdown-submenu">
    <a tabindex="-1" href="">Comparative Genomics</a>
    <ul class="dropdown-menu">
        
            
<li >
    <a href="../1a/">Intro to Galaxy</a>
</li>

        
            
<li >
    <a href="../1b/">Intro to Artemis</a>
</li>

        
            
<li >
    <a href="../2a/">Assembly with Spades</a>
</li>

        
            
<li >
    <a href="../2b/">Annotation with Prokka</a>
</li>

        
            
<li class="active">
    <a href="./">Comparative genomics</a>
</li>

        
    </ul>
  </li>

                    
                        
<li >
    <a href="../../snps/">Variant calling</a>
</li>

                    
                        
<li >
    <a href="../../trees/">Phylogenetic trees</a>
</li>

                    
                        
<li >
    <a href="../../dge/">Differential gene expression</a>
</li>

                    
                    </ul>
                </li>
            
            
            
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Data <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        
<li >
    <a href="../../../data/data_access/">Getting and moving data</a>
</li>

                    
                        
<li >
    <a href="../../../data/formats/file_formats/">File formats</a>
</li>

                    
                        
<li >
    <a href="../../../dna/data/">Genomics</a>
</li>

                    
                        
<li >
    <a href="../../../rna/data/">Transcriptomics</a>
</li>

                    
                        
<li >
    <a href="../../../pro/data/">Proteomics</a>
</li>

                    
                        
<li >
    <a href="../../../met/data/">Metabolomics</a>
</li>

                    
                    </ul>
                </li>
            
            
            
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Tools <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        
  <li class="dropdown-submenu">
    <a tabindex="-1" href="">Genomics</a>
    <ul class="dropdown-menu">
        
            
  <li class="dropdown-submenu">
    <a tabindex="-1" href="">Quality Control</a>
    <ul class="dropdown-menu">
        
            
<li >
    <a href="../../../dna/qualitycontrol/qc/">Overview</a>
</li>

        
            
<li >
    <a href="../../../dna/qualitycontrol/fastqc/">FastQC</a>
</li>

        
            
<li >
    <a href="../../../dna/qualitycontrol/fastqc_cmdline/">FastQC commandline</a>
</li>

        
            
<li >
    <a href="../../../dna/qualitycontrol/trimmomatic/">Trimmomatic</a>
</li>

        
    </ul>
  </li>

        
            
  <li class="dropdown-submenu">
    <a tabindex="-1" href="">De novo assembly</a>
    <ul class="dropdown-menu">
        
            
<li >
    <a href="../../../dna/denovo/spades/">Spades</a>
</li>

        
            
<li >
    <a href="../../../dna/denovo/megahit/">Megahit</a>
</li>

        
            
<li >
    <a href="../../../dna/denovo/velvet/">Velvet</a>
</li>

        
            
<li >
    <a href="../../../dna/denovo/vopt/">VelvetOptimiser</a>
</li>

        
            
<li >
    <a href="../../../dna/denovo/pacbio/">PacBio SMRT portal</a>
</li>

        
            
<li >
    <a href="../../../dna/denovo/galaxy-spades/">Galaxy-Spades</a>
</li>

        
    </ul>
  </li>

        
            
  <li class="dropdown-submenu">
    <a tabindex="-1" href="">Annotation</a>
    <ul class="dropdown-menu">
        
            
<li >
    <a href="../../../dna/anno/prokka/">Prokka</a>
</li>

        
            
<li >
    <a href="../../../dna/anno/rast/">RAST</a>
</li>

        
            
<li >
    <a href="../../../dna/anno/genix/">GENIX</a>
</li>

        
            
<li >
    <a href="../../../dna/anno/eugene/">EuGene</a>
</li>

        
            
<li >
    <a href="../../../dna/anno/megannotator/">MEGAnnotator</a>
</li>

        
            
<li >
    <a href="../../../dna/anno/prokka_cmdline/">Prokka on cmdline</a>
</li>

        
    </ul>
  </li>

        
            
  <li class="dropdown-submenu">
    <a tabindex="-1" href="">Variant finding</a>
    <ul class="dropdown-menu">
        
            
<li >
    <a href="../../../dna/snps/background/background/">Background</a>
</li>

        
            
<li >
    <a href="../../../dna/snps/snippy/">Snippy</a>
</li>

        
            
<li >
    <a href="../../../dna/snps/galaxy/">Galaxy</a>
</li>

        
            
<li >
    <a href="../../../dna/snps/bwa_mem/">BWA-MEM</a>
</li>

        
    </ul>
  </li>

        
            
  <li class="dropdown-submenu">
    <a tabindex="-1" href="">Pan genome</a>
    <ul class="dropdown-menu">
        
            
<li >
    <a href="../../../dna/pan/roary/">Roary</a>
</li>

        
            
<li >
    <a href="../../../dna/pan/orthomcl/">OrthoMCL</a>
</li>

        
            
<li >
    <a href="../../../dna/pan/mauve/">Mauve</a>
</li>

        
    </ul>
  </li>

        
    </ul>
  </li>

                    
                        
  <li class="dropdown-submenu">
    <a tabindex="-1" href="">Transcriptomics</a>
    <ul class="dropdown-menu">
        
            
<li >
    <a href="../../../rna/qc/">Quality Control</a>
</li>

        
            
  <li class="dropdown-submenu">
    <a tabindex="-1" href="">De novo assembly</a>
    <ul class="dropdown-menu">
        
            
<li >
    <a href="../../../rna/denovo/oases/">Oases</a>
</li>

        
            
<li >
    <a href="../../../rna/denovo/trinity/">Trinity</a>
</li>

        
            
<li >
    <a href="../../../rna/denovo/stringtie/">StringTie</a>
</li>

        
    </ul>
  </li>

        
    </ul>
  </li>

                    
                        
  <li class="dropdown-submenu">
    <a tabindex="-1" href="">Proteomics</a>
    <ul class="dropdown-menu">
        
            
<li >
    <a href="../../../pro/data/">The Data</a>
</li>

        
            
<li >
    <a href="../../../pro/qc/">Quality Control</a>
</li>

        
    </ul>
  </li>

                    
                        
  <li class="dropdown-submenu">
    <a tabindex="-1" href="">Metabolomics</a>
    <ul class="dropdown-menu">
        
            
<li >
    <a href="../../../met/data/">The Data</a>
</li>

        
            
<li >
    <a href="../../../met/qc/">Quality Control</a>
</li>

        
    </ul>
  </li>

                    
                    </ul>
                </li>
            
            
            
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Viz <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        
<li >
    <a href="../../../viz/bandage/">Bandage</a>
</li>

                    
                        
<li >
    <a href="../../../viz/jbrowse/">JBrowse</a>
</li>

                    
                        
<li >
    <a href="../../../viz/artemis/">Artemis</a>
</li>

                    
                        
<li >
    <a href="../../../viz/phandango/">Phandango</a>
</li>

                    
                        
<li >
    <a href="../../../viz/degust/">Degust</a>
</li>

                    
                    </ul>
                </li>
            
            
            
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Protocols <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        
<li >
    <a href="../../../protocols/overview/">Overview</a>
</li>

                    
                        
<li >
    <a href="../../../protocols/assembly/">De novo assembly</a>
</li>

                    
                        
<li >
    <a href="../../../protocols/annotation/">Genome annotation</a>
</li>

                    
                        
<li >
    <a href="../../../protocols/variants/">Finding variants</a>
</li>

                    
                        
<li >
    <a href="../../../protocols/dge/">Differential gene expression</a>
</li>

                    
                    </ul>
                </li>
            
            
            
                <li >
                    <a href="../../../about/">About</a>
                </li>
            
            
            </ul>

            <!-- Search, Navigation and Repo links -->
            <ul class="nav navbar-nav navbar-right">
                <li>
                    <a href="#" data-toggle="modal" data-target="#mkdocs_search_modal">
                        <i class="fa fa-search"></i> Search
                    </a>
                </li>
                <li >
                    <a rel="next" href="../2b/">
                        <i class="fa fa-arrow-left"></i> Previous
                    </a>
                </li>
                <li >
                    <a rel="prev" href="../../snps/">
                        Next <i class="fa fa-arrow-right"></i>
                    </a>
                </li>
                
            </ul>
        </div>
    </div>
</div>

        <div class="container">
            <div class="col-md-3"><div class="bs-sidebar hidden-print affix well" role="complementary">
    <ul class="nav bs-sidenav">
    
        <li class="main active"><a href="#comparative-genomics">Comparative Genomics</a></li>
        
            <li><a href="#background">Background</a></li>
        
            <li><a href="#learning-objectives">Learning objectives</a></li>
        
            <li><a href="#input-data">Input data</a></li>
        
            <li><a href="#map-reads-to-reference">Map reads to reference</a></li>
        
            <li><a href="#view-the-bam-file-using-artemis">View the BAM file using Artemis</a></li>
        
            <li><a href="#variant-calling">Variant Calling</a></li>
        
            <li><a href="#investigation-of-variants">Investigation of Variants</a></li>
        
            <li><a href="#detection-of-macro-indels-and-rearrangement-using-mauve">Detection of ‘macro’ INDELS and rearrangement using Mauve</a></li>
        
    
    </ul>
</div></div>
            <div class="col-md-9" role="main">

<h1 id="comparative-genomics">Comparative Genomics</h1>
<h2 id="background">Background</h2>
<p>In this activity we will identify ‘micro’ differences between genome
sequences using the BWA short read mapper and Freebayes variant caller. After investigating the ‘micro’ differences (SNPs/INDELS), we will attempt to detect larger ‘macro’ differences using Mauve.</p>
<h2 id="learning-objectives">Learning objectives</h2>
<p>At the end of this tutorial you should be able to:</p>
<ol>
<li>map sequence reads to a reference genome</li>
<li>view the mapped reads</li>
<li>identify variants using Freebayes, and</li>
<li>identify larger structural variants using Mauve.</li>
</ol>
<h2 id="input-data">Input data</h2>
<p>The genome sequences being compared are those of the ‘wildtype’ and ‘mutant’ strains.</p>
<p>The relevant files should already be available on Galaxy (from the previous &ldquo;Assembly with Spades&rdquo; tutorial).</p>
<p>Just for a recap:</p>
<p>We have a closed, annotated genome sequence for the wildtype strain. This file has two formats (one with the sequence, and one with the features).</p>
<ul>
<li>Files: <fn>wildtype.fna</fn> and <fn>wildtype.gff</fn></li>
</ul>
<p>For the mutant strain we have whole genome shotgun short sequence reads
from an Illumina DNA sequencing instrument.</p>
<ul>
<li>
<p>Files: <fn>mutant_R1.fastq</fn> and <fn>mutant_R2.fastq</fn> (fastq format)</p>
</li>
<li>
<p>The reads are paired-end</p>
</li>
<li>Each read is 150 bases  </li>
<li>The reads coverage depth is estimated at 19x.</li>
</ul>
<h2 id="map-reads-to-reference">Map reads to reference</h2>
<h3 id="map-the-reads-on-to-the-reference-sequence">Map the reads on to the reference sequence</h3>
<p>Several programs could be used for this but we will use BWA-MEM.</p>
<ul>
<li>Go to <ss>Tools &rarr; NGS Analysis &rarr; NGS: Mapping &rarr; Map with BWA-MEM</ss>.</li>
<li>Set the following parameters:</li>
<li><ss>Will you select a reference genome from your history or use a built-in index?</ss>: <em>Use a genome from history and build index</em></li>
<li><ss>Use the following dataset as the reference sequence</ss>: <em>wildtype.fna</em></li>
<li><ss>Select first set of reads</ss>:<em>mutant_R1.fastq</em></li>
<li><ss>Select second set of reads</ss>:<em>mutant_R2.fastq</em></li>
<li>Click <ss>Execute</ss>.</li>
</ul>
<h3 id="examine-the-mapped-reads">Examine the mapped reads</h3>
<p>To do this, we will look at the contents of the BAM file.</p>
<div class="admonition hint">
<p class="admonition-title">Hint</p>
<p>The BAM file is a Binary Compressed Datafile and cannot be viewed directly. If you attempt to view it using the &ldquo;view data&rdquo; button (the eye icon) it will be downloaded to your local computer. Instead, we must convert it to a non-compressed text format (SAM) first.</p>
</div>
<ul>
<li>Go to <ss>Tools &rarr; NGS Common Toolsets &rarr; NGS: SAM tools &rarr; BAM-to-SAM</ss>.</li>
<li><ss>BAM File to Convert</ss>: <em>your BAM file</em></li>
<li><ss>Execute</ss>.</li>
<li>View the resultant SAM file by clicking on the eye icon.</li>
<li>Have a look at the fields in the file (the column headings).</li>
<li>The demonstrator will now point out what all the fields are. (FIXME: or add some info here?)</li>
</ul>
<h2 id="view-the-bam-file-using-artemis">View the BAM file using Artemis</h2>
<p>In this section we will use Artemis to view the BAM file we produced above.</p>
<h3 id="download-the-bam-file-to-your-local-computer">Download the BAM file to your local computer.</h3>
<ul>
<li>Click on the name of the BAM file that you created in Section 1.  </li>
<li>Click on the download button <img alt="download icon" src="./images/image02.png" />; you need to download both the BAM file and the bam_index. (FIXME: why?)</li>
</ul>
<p><img alt="screenshot of file download" src="./images/image03.png" /></p>
<ul>
<li>Also, download the annotated reference sequence; <fn>wildtype.gff</fn>.</li>
</ul>
<h3 id="start-artemis-and-load-the-wildtypegff">Start Artemis and load the wildtype.gff</h3>
<ul>
<li>From the Artemis menu, Click <ss>File &rarr; Open &hellip;</ss>  </li>
<li>Load <fn>wildtype.gff</fn>.</li>
</ul>
<p>You should now have the wildtype’s annotated sequence loaded into the Artemis genome browser.</p>
<h3 id="load-the-bam-file-into-artemis">Load the BAM file into Artemis</h3>
<ul>
<li>Click <ss>File &rarr; Read BAM/VCF</ss>  </li>
<li><ss>Select</ss>: <fn>Galaxy … .bam</fn>  </li>
<li>Click <ss>Ok</ss>   </li>
</ul>
<p>FIXME: but don&rsquo;t load index? bai file?</p>
<p>You should see something like this:</p>
<p><img alt="BAM file in Artemis" src="./images/image00.png" /></p>
<p>Can you find a SNP?</p>
<p>Demonstration of the ways that the view of the BAM file can be enhanced!</p>
<p>Imagine finding each SNP manually - luckily this can be automated using a tool available on Galaxy.</p>
<h2 id="variant-calling">Variant Calling</h2>
<p>We will now search through our alignment file (BAM) for statistically-valid SNPs using the Freebayes variant calling tool.</p>
<h3 id="run-freebayes">Run Freebayes</h3>
<ul>
<li>Go to <ss>Tools &rarr; NGS Analysis &rarr; NGS: Variant Analysis &rarr; FreeBayes</ss></li>
<li>Set the following parameters:</li>
<li><ss>Load reference genome from</ss>: <em>History</em>  </li>
<li><ss>Sample BAM file</ss>: <em>Map with BWA-MEM on data … BAM format</em></li>
<li><ss>Use the following dataset as the reference sequence</ss>: <fn>wildtype.fna</fn>  </li>
<li>Click <ss>Execute</ss></li>
</ul>
<h3 id="examine-the-freebayes-output">Examine the Freebayes output</h3>
<ul>
<li>Freebayes will create a VCF file. This stands for Variant Calling Format.</li>
<li>Click on its View Data button (eye icon) and have a look at the file. There is a lot of header information; the variants appear lower down.</li>
<li>Can you spot a SNP?</li>
<li>What about an insertion? A deletion?</li>
</ul>
<h2 id="investigation-of-variants">Investigation of Variants</h2>
<p>What is the impact of the differences we have observed?</p>
<p>In this section we will use some simple strategies to predict the impact of the variant on the function of the gene and
perhaps even the strain itself.</p>
<p><strong>Artemis</strong> - the annotated draft genome sequence of the mutant strain -
what is the impact the protein coding region? what is the predicted
function?</p>
<p><strong>blastp</strong> - <a href="http://blast.ncbi.nlm.nih.gov/Blast.cgi">http://blast.ncbi.nlm.nih.gov/Blast.cgi</a> the protein domain display - are any major protein domains truncated by the difference?</p>
<p><strong>LipoP/SignalP/TmHMM</strong> - <a href="http://www.cbs.dtu.dk/services/">http://www.cbs.dtu.dk/services/</a> membrane location prediction - has the change had an impact on the membrane location of the protein?</p>
<p><strong>Literature?</strong></p>
<p>Can you suggest a type of nucleotide sequence that might have no impact on the function of the encoded protein?</p>
<p>In this section we will investigate a few variants together as a demonstration</p>
<p>perhaps a few individually too??</p>
<h2 id="detection-of-macro-indels-and-rearrangement-using-mauve">Detection of ‘macro’ INDELS and rearrangement using Mauve</h2>
<p>We will now examine our earlier assembly and compare it with the reference on a genome wide basis using Mauve.</p>
<p>Download and install Mauve. More information on Mauve and its use can be found <a href="http://darlinglab.org/mauve/mauve.html">here</a>.</p>
<p>You will then need to load both the reference <fn>wildtype.gff</fn> file and the <fn>mutant.gff</fn> file that you downloaded earlier.</p>
<p><img alt="Mauve screenshot" src="./images/image01.png" /></p>
<p>FIXME: add more</p></div>
        </div>

        <footer class="col-md-12">
            <hr>
            
            <center>Documentation built with <a href="http://www.mkdocs.org/">MkDocs</a>.</center>
        </footer>

        <script src="../../../js/jquery-1.10.2.min.js"></script>
        <script src="../../../js/bootstrap-3.0.3.min.js"></script>
        <script src="../../../js/highlight.pack.js"></script>
        <script>var base_url = '../../..';</script>
        <script data-main="../../../mkdocs/js/search.js" src="../../../mkdocs/js/require.js"></script>
        <script src="../../../js/base.js"></script>

        <div class="modal" id="mkdocs_search_modal" tabindex="-1" role="dialog" aria-labelledby="Search Modal" aria-hidden="true">
            <div class="modal-dialog">
                <div class="modal-content">
                    <div class="modal-header">
                        <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
                        <h4 class="modal-title" id="exampleModalLabel">Search</h4>
                    </div>
                    <div class="modal-body">
                        <p>
                            From here you can search these documents. Enter
                            your search terms below.
                        </p>
                        <form role="form">
                            <div class="form-group">
                                <input type="text" class="form-control" placeholder="Search..." id="mkdocs-search-query">
                            </div>
                        </form>
                        <div id="mkdocs-search-results"></div>
                    </div>
                    <div class="modal-footer">
                    </div>
                </div>
            </div>
        </div>
    </body>
</html>