<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        
        <meta name="author" content="Torsten Seemann, Anna Syme">
        
        <link rel="shortcut icon" href="../../../favicon.ico">
        

	<title>Roary - Sepsis Tutorials</title>

        <link href="../../../css/bootstrap-custom.min.css" rel="stylesheet">
        <link href="../../../css/font-awesome-4.0.3.css" rel="stylesheet">
        <link rel="stylesheet" href="../../../css/highlight.css">
        <link href="../../../css/base.css" rel="stylesheet">
        <link href="../../../css/sepsis.css" rel="stylesheet">

        <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
        <!--[if lt IE 9]>
            <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
            <script src="https://oss.maxcdn.com/libs/respond.js/1.3.0/respond.min.js"></script>
        <![endif]-->

        
    </head>

    <body>

        <div class="navbar navbar-default navbar-fixed-top" role="navigation">
    <div class="container">

        <!-- Collapsed navigation -->
        <div class="navbar-header">
            <!-- Expander button -->
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>

            <!-- Main title -->
            <a class="navbar-brand" href="../../..">Sepsis Tutorials</a>
        </div>

        <!-- Expanded navigation -->
        <div class="navbar-collapse collapse">
            <!-- Main navigation -->
            <ul class="nav navbar-nav">
            
            
                <li >
                    <a href="../../..">Home</a>
                </li>
            
            
            
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Workshops <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        
  <li class="dropdown-submenu">
    <a tabindex="-1" href="">Comparative Genomics</a>
    <ul class="dropdown-menu">
        
            
<li >
    <a href="../../../workshop/genomics/1a/">Intro to Galaxy</a>
</li>

        
            
<li >
    <a href="../../../workshop/genomics/1b/">Intro to Artemis</a>
</li>

        
            
<li >
    <a href="../../../workshop/genomics/2a/">Assembly with Spades</a>
</li>

        
            
<li >
    <a href="../../../workshop/genomics/2b/">Annotation with Prokka</a>
</li>

        
            
<li >
    <a href="../../../workshop/genomics/3/">Comparative genomics</a>
</li>

        
    </ul>
  </li>

                    
                        
<li >
    <a href="../../../workshop/snps/">Variant calling</a>
</li>

                    
                        
<li >
    <a href="../../../workshop/trees/">Phylogenetic trees</a>
</li>

                    
                        
<li >
    <a href="../../../workshop/dge/">Differential gene expression</a>
</li>

                    
                    </ul>
                </li>
            
            
            
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Data <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        
<li >
    <a href="../../../data/data_access/">Getting and moving data</a>
</li>

                    
                        
<li >
    <a href="../../../data/formats/file_formats/">File formats</a>
</li>

                    
                        
<li >
    <a href="../../data/">Genomics</a>
</li>

                    
                        
<li >
    <a href="../../../rna/data/">Transcriptomics</a>
</li>

                    
                        
<li >
    <a href="../../../pro/data/">Proteomics</a>
</li>

                    
                        
<li >
    <a href="../../../met/data/">Metabolomics</a>
</li>

                    
                    </ul>
                </li>
            
            
            
                <li class="dropdown active">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Tools <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        
  <li class="dropdown-submenu">
    <a tabindex="-1" href="">Genomics</a>
    <ul class="dropdown-menu">
        
            
  <li class="dropdown-submenu">
    <a tabindex="-1" href="">Quality Control</a>
    <ul class="dropdown-menu">
        
            
<li >
    <a href="../../qualitycontrol/qc/">Overview</a>
</li>

        
            
<li >
    <a href="../../qualitycontrol/fastqc/">FastQC</a>
</li>

        
            
<li >
    <a href="../../qualitycontrol/fastqc_cmdline/">FastQC commandline</a>
</li>

        
            
<li >
    <a href="../../qualitycontrol/trimmomatic/">Trimmomatic</a>
</li>

        
    </ul>
  </li>

        
            
  <li class="dropdown-submenu">
    <a tabindex="-1" href="">De novo assembly</a>
    <ul class="dropdown-menu">
        
            
<li >
    <a href="../../denovo/spades/">Spades</a>
</li>

        
            
<li >
    <a href="../../denovo/megahit/">Megahit</a>
</li>

        
            
<li >
    <a href="../../denovo/velvet/">Velvet</a>
</li>

        
            
<li >
    <a href="../../denovo/vopt/">VelvetOptimiser</a>
</li>

        
            
<li >
    <a href="../../denovo/pacbio/">PacBio SMRT portal</a>
</li>

        
            
<li >
    <a href="../../denovo/galaxy-spades/">Galaxy-Spades</a>
</li>

        
    </ul>
  </li>

        
            
  <li class="dropdown-submenu">
    <a tabindex="-1" href="">Annotation</a>
    <ul class="dropdown-menu">
        
            
<li >
    <a href="../../anno/prokka/">Prokka</a>
</li>

        
            
<li >
    <a href="../../anno/rast/">RAST</a>
</li>

        
            
<li >
    <a href="../../anno/genix/">GENIX</a>
</li>

        
            
<li >
    <a href="../../anno/eugene/">EuGene</a>
</li>

        
            
<li >
    <a href="../../anno/megannotator/">MEGAnnotator</a>
</li>

        
            
<li >
    <a href="../../anno/prokka_cmdline/">Prokka on cmdline</a>
</li>

        
    </ul>
  </li>

        
            
  <li class="dropdown-submenu">
    <a tabindex="-1" href="">Variant finding</a>
    <ul class="dropdown-menu">
        
            
<li >
    <a href="../../snps/background/background/">Background</a>
</li>

        
            
<li >
    <a href="../../snps/snippy/">Snippy</a>
</li>

        
            
<li >
    <a href="../../snps/galaxy/">Galaxy</a>
</li>

        
            
<li >
    <a href="../../snps/bwa_mem/">BWA-MEM</a>
</li>

        
    </ul>
  </li>

        
            
  <li class="dropdown-submenu">
    <a tabindex="-1" href="">Pan genome</a>
    <ul class="dropdown-menu">
        
            
<li class="active">
    <a href="./">Roary</a>
</li>

        
            
<li >
    <a href="../orthomcl/">OrthoMCL</a>
</li>

        
            
<li >
    <a href="../mauve/">Mauve</a>
</li>

        
    </ul>
  </li>

        
    </ul>
  </li>

                    
                        
  <li class="dropdown-submenu">
    <a tabindex="-1" href="">Transcriptomics</a>
    <ul class="dropdown-menu">
        
            
<li >
    <a href="../../../rna/qc/">Quality Control</a>
</li>

        
            
  <li class="dropdown-submenu">
    <a tabindex="-1" href="">De novo assembly</a>
    <ul class="dropdown-menu">
        
            
<li >
    <a href="../../../rna/denovo/oases/">Oases</a>
</li>

        
            
<li >
    <a href="../../../rna/denovo/trinity/">Trinity</a>
</li>

        
            
<li >
    <a href="../../../rna/denovo/stringtie/">StringTie</a>
</li>

        
    </ul>
  </li>

        
    </ul>
  </li>

                    
                        
  <li class="dropdown-submenu">
    <a tabindex="-1" href="">Proteomics</a>
    <ul class="dropdown-menu">
        
            
<li >
    <a href="../../../pro/data/">The Data</a>
</li>

        
            
<li >
    <a href="../../../pro/qc/">Quality Control</a>
</li>

        
    </ul>
  </li>

                    
                        
  <li class="dropdown-submenu">
    <a tabindex="-1" href="">Metabolomics</a>
    <ul class="dropdown-menu">
        
            
<li >
    <a href="../../../met/data/">The Data</a>
</li>

        
            
<li >
    <a href="../../../met/qc/">Quality Control</a>
</li>

        
    </ul>
  </li>

                    
                    </ul>
                </li>
            
            
            
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Viz <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        
<li >
    <a href="../../../viz/bandage/">Bandage</a>
</li>

                    
                        
<li >
    <a href="../../../viz/jbrowse/">JBrowse</a>
</li>

                    
                        
<li >
    <a href="../../../viz/artemis/">Artemis</a>
</li>

                    
                        
<li >
    <a href="../../../viz/phandango/">Phandango</a>
</li>

                    
                        
<li >
    <a href="../../../viz/degust/">Degust</a>
</li>

                    
                    </ul>
                </li>
            
            
            
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Protocols <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        
<li >
    <a href="../../../protocols/overview/">Overview</a>
</li>

                    
                        
<li >
    <a href="../../../protocols/assembly/">De novo assembly</a>
</li>

                    
                        
<li >
    <a href="../../../protocols/annotation/">Genome annotation</a>
</li>

                    
                        
<li >
    <a href="../../../protocols/variants/">Finding variants</a>
</li>

                    
                        
<li >
    <a href="../../../protocols/dge/">Differential gene expression</a>
</li>

                    
                    </ul>
                </li>
            
            
            
                <li >
                    <a href="../../../about/">About</a>
                </li>
            
            
            </ul>

            <!-- Search, Navigation and Repo links -->
            <ul class="nav navbar-nav navbar-right">
                <li>
                    <a href="#" data-toggle="modal" data-target="#mkdocs_search_modal">
                        <i class="fa fa-search"></i> Search
                    </a>
                </li>
                <li >
                    <a rel="next" href="../../snps/bwa_mem/">
                        <i class="fa fa-arrow-left"></i> Previous
                    </a>
                </li>
                <li >
                    <a rel="prev" href="../orthomcl/">
                        Next <i class="fa fa-arrow-right"></i>
                    </a>
                </li>
                
            </ul>
        </div>
    </div>
</div>

        <div class="container">
            <div class="col-md-3"><div class="bs-sidebar hidden-print affix well" role="complementary">
    <ul class="nav bs-sidenav">
    
        <li class="main active"><a href="#roary">Roary</a></li>
        
            <li><a href="#pre-requisites">Pre-requisites</a></li>
        
            <li><a href="#start">Start</a></li>
        
            <li><a href="#input-data">Input data</a></li>
        
            <li><a href="#run">Run</a></li>
        
            <li><a href="#how-it-works">How it works</a></li>
        
            <li><a href="#output">Output</a></li>
        
            <li><a href="#advanced-options">Advanced options</a></li>
        
            <li><a href="#what-next">What next?</a></li>
        
            <li><a href="#more-information">More information</a></li>
        
    
    </ul>
</div></div>
            <div class="col-md-9" role="main">

<h1 id="roary">Roary</h1>
<p>cmdline</p>
<p>This tutorial demonstrates how to calculate the pan and core genomes of a set of input bacterial species, using Roary.</p>
<p><a href="https://sanger-pathogens.github.io/Roary/">Roary code and manual on github</a></p>
<p><a href="http://bioinformatics.oxfordjournals.org/content/31/22/3691">Roary paper</a></p>
<h2 id="pre-requisites">Pre-requisites</h2>
<ul>
<li>background: pan genomes</li>
<li>a mGVL instance</li>
</ul>
<h2 id="start">Start</h2>
<ul>
<li><strong>via local Terminal</strong>: in your terminal, ssh into your mGVL, but make sure you put in -X -Y after ssh so that xquartz can view files later. (FIXME: word better)</li>
<li><strong>or via virtual desktop</strong>: Go to your mGVL dashboard. Click on the link to the Lubuntu desktop. A virtual desktop will open in a new browser window. Enter username: ubunutu; and your GVL password. Click on terminal in the top left corner.</li>
</ul>
<h2 id="input-data">Input data</h2>
<ul>
<li>Roary takes .gff files produced by Prokka. A gff file has sequences and annotations.</li>
<li><a href="../../dna/anno/prokka">file formats</a> FIXME: link to proper page</li>
<li>Get files into mGVL. (FIXME: explain how, wget etc or from GenomeSpace)</li>
<li>put all gff files into a folder</li>
<li>FIXME: choose a good sample set - ideally something that usefully shows how AMR genes can be present/absence in a group? for a draft can use the Listeria tutorial at https://github.com/microgenomics/tutorials/blob/master/pangenome.md</li>
</ul>
<h2 id="run">Run</h2>
<ul>
<li>navigate into the place where the gff folder is.</li>
</ul>
<pre><code class="bash">roary -f ./results ./gff_files/*.gff
</code></pre>

<ul>
<li>&ldquo;-f ./results&rdquo; puts the output into a directory called results</li>
</ul>
<h2 id="how-it-works">How it works</h2>
<ul>
<li>Based on the input genomes, Roary works out which genes are shared between all (core) and which are not (accessory).</li>
<li>It uses the protein-coding genes from each of the input genomes.</li>
<li>converts to protein seqs</li>
<li>similar protein seqs are clustered progressively.</li>
<li>each sample: will be labelled with presence/absence of orthologous genes.</li>
</ul>
<h2 id="output">Output</h2>
<h3 id="summary-statistics">summary statistics:</h3>
<pre><code>more summary_statistics.txt
</code></pre>

<ul>
<li>you will see the number of core genes, shell genes, etc.</li>
<li><code>q</code> to exit viewing</li>
</ul>
<h3 id="gene-presenceabsence-graphically">gene presence/absence graphically:</h3>
<pre><code>roary2svg.pl gene_presence_absence.csv &gt; pan_genome.svg
</code></pre>

<ul>
<li>(if you have logged in with -X -Y)</li>
</ul>
<pre><code>firefox pan_genome.svg &amp;
</code></pre>

<ul>
<li>then <code>enter</code></li>
<li>the &amp; makes it run in the background</li>
<li>a firefox window should open with the svg image</li>
<li>(later: close the firefox window to stop this job running in the background)</li>
</ul>
<h3 id="list-of-genes-that-are-presentabsent">list of genes that are present/absent:</h3>
<ul>
<li>view the gene_presence_absence.csv by (FIXME)</li>
<li>lots of information about this file in the roary website (FIXME summarize?)</li>
</ul>
<h3 id="query-the-pan-genome">query the pan genome:</h3>
<ul>
<li>copy the input .gff files into the results folder (FIXME: do this earlier)</li>
<li>cd into this folder</li>
</ul>
<pre><code>query_pan_genome -a intersection *.gff
</code></pre>

<ul>
<li>this finds the core genes</li>
</ul>
<pre><code>more pan_genome_results
</code></pre>

<ul>
<li>shows the list of genes found in the core genome.</li>
<li><code>q</code> to exit viewing</li>
</ul>
<h2 id="advanced-options">Advanced options</h2>
<p>FIXME: update firefox on mGVL so can run phandango</p>
<h3 id="run-roary-and-create-an-alignment-of-core-genes">Run roary and create an alignment of core genes:</h3>
<pre><code class="bash">roary -f ./results -e -n -p 8 ./gff_files/*.gff
</code></pre>

<ul>
<li>&ldquo;-f ./results&rdquo; puts the output into a directory called results</li>
<li>&ldquo;-e -n&rdquo; creates an alignment of core genes using mafft</li>
<li>&ldquo;-p 8&rdquo; gives 8 threads - optional, if you know how many you have</li>
</ul>
<h3 id="generate-a-tree-based-on-the-presenceabsence-of-core-genes">Generate a tree based on the presence/absence of core genes:</h3>
<ul>
<li>navigate into the results folder that you want to use.</li>
</ul>
<pre><code class="bash">FastTree -nt -gtr core_gene_alignment.aln &gt; my_tree.newick
</code></pre>

<ul>
<li>(By default, roary will also have created a (very quick) tree from the accessory genes.)</li>
<li><a href="http://meta.microbesonline.org/fasttree/">FastTree information and options</a>.</li>
</ul>
<h3 id="use-roary_plotspy-to-generate-plots">Use roary_plots.py to generate plots:</h3>
<ul>
<li>navigate into the results folder that you want to use.</li>
</ul>
<pre><code>python roary_plots.py core_gene_alignment.nwk gene_presence_absence.csv
</code></pre>

<ul>
<li>output: pangenome matrix, frequency plot, pie chart.</li>
<li>view these by typing <code>firefox [filename]</code> and a firefox window will open to show the image. You need to close the window before you open the next image.</li>
</ul>
<h2 id="what-next">What next?</h2>
<p>View using Phandango; <a href="../../../viz/phandango/">tutorial here.</a></p>
<h2 id="more-information">More information</h2>
<ul>
<li><a href="https://github.com/microgenomics/tutorials/blob/master/pangenome.md">another Roary tutorial</a></li>
</ul></div>
        </div>

        <footer class="col-md-12">
            <hr>
            
            <center>Documentation built with <a href="http://www.mkdocs.org/">MkDocs</a>.</center>
        </footer>

        <script src="../../../js/jquery-1.10.2.min.js"></script>
        <script src="../../../js/bootstrap-3.0.3.min.js"></script>
        <script src="../../../js/highlight.pack.js"></script>
        <script>var base_url = '../../..';</script>
        <script data-main="../../../mkdocs/js/search.js" src="../../../mkdocs/js/require.js"></script>
        <script src="../../../js/base.js"></script>

        <div class="modal" id="mkdocs_search_modal" tabindex="-1" role="dialog" aria-labelledby="Search Modal" aria-hidden="true">
            <div class="modal-dialog">
                <div class="modal-content">
                    <div class="modal-header">
                        <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
                        <h4 class="modal-title" id="exampleModalLabel">Search</h4>
                    </div>
                    <div class="modal-body">
                        <p>
                            From here you can search these documents. Enter
                            your search terms below.
                        </p>
                        <form role="form">
                            <div class="form-group">
                                <input type="text" class="form-control" placeholder="Search..." id="mkdocs-search-query">
                            </div>
                        </form>
                        <div id="mkdocs-search-results"></div>
                    </div>
                    <div class="modal-footer">
                    </div>
                </div>
            </div>
        </div>
    </body>
</html>