<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        
        <meta name="author" content="Torsten Seemann, Anna Syme">
        
        <link rel="shortcut icon" href="../../../favicon.ico">
        

	<title>Snippy on commandline - Sepsis Tutorials</title>

        <link href="../../../css/bootstrap-custom.min.css" rel="stylesheet">
        <link href="../../../css/font-awesome-4.0.3.css" rel="stylesheet">
        <link rel="stylesheet" href="../../../css/highlight.css">
        <link href="../../../css/base.css" rel="stylesheet">
        <link href="../../../css/sepsis.css" rel="stylesheet">

        <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
        <!--[if lt IE 9]>
            <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
            <script src="https://oss.maxcdn.com/libs/respond.js/1.3.0/respond.min.js"></script>
        <![endif]-->

        
    </head>

    <body>

        <div class="navbar navbar-default navbar-fixed-top" role="navigation">
    <div class="container">

        <!-- Collapsed navigation -->
        <div class="navbar-header">
            <!-- Expander button -->
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>

            <!-- Main title -->
            <a class="navbar-brand" href="../../..">Sepsis Tutorials</a>
        </div>

        <!-- Expanded navigation -->
        <div class="navbar-collapse collapse">
            <!-- Main navigation -->
            <ul class="nav navbar-nav">
            
            
                <li >
                    <a href="../../..">Home</a>
                </li>
            
            
            
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Introductory training <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        
<li >
    <a href="../../../workshop/overview/">Overview</a>
</li>

                    
                        
<li >
    <a href="../../../workshop/genomics/1a/">Use Galaxy</a>
</li>

                    
                        
<li >
    <a href="../../../workshop/genomics/1c/">View with JBroswe</a>
</li>

                    
                        
<li >
    <a href="../../../workshop/genomics/2a/">Assemble with Spades</a>
</li>

                    
                        
<li >
    <a href="../../../workshop/genomics/2b/">Annotate with Prokka</a>
</li>

                    
                        
<li >
    <a href="../../../workshop/genomics/3b/">Find variants with Snippy</a>
</li>

                    
                    </ul>
                </li>
            
            
            
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Advanced training <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        
<li >
    <a href="../../../how_to/overview/">Overview</a>
</li>

                    
                        
<li >
    <a href="../../../how_to/assembly/assembly_galaxy/">Genome assembly in Galaxy</a>
</li>

                    
                        
<li >
    <a href="../../../how_to/annotation/anno_galaxy/">Genome annotation in Galaxy</a>
</li>

                    
                        
<li >
    <a href="../../../how_to/pan/roary/">Pan genomes</a>
</li>

                    
                    </ul>
                </li>
            
            
            
                <li class="dropdown active">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Tools <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        
<li >
    <a href="../../../tools/">Oveview</a>
</li>

                    
                        
  <li class="dropdown-submenu">
    <a tabindex="-1" href="">Genomics</a>
    <ul class="dropdown-menu">
        
            
  <li class="dropdown-submenu">
    <a tabindex="-1" href="">Quality Control</a>
    <ul class="dropdown-menu">
        
            
<li >
    <a href="../../qualitycontrol/fastqc/">FastQC in Galaxy</a>
</li>

        
            
<li >
    <a href="../../qualitycontrol/fastqc_cmdline/">FastQC on commandline</a>
</li>

        
            
<li >
    <a href="../../qualitycontrol/trimmomatic/">Trimmomatic in Galaxy</a>
</li>

        
    </ul>
  </li>

        
            
  <li class="dropdown-submenu">
    <a tabindex="-1" href="">De novo assembly</a>
    <ul class="dropdown-menu">
        
            
<li >
    <a href="../../denovo/pacbio/">PacBio SMRT portal</a>
</li>

        
            
<li >
    <a href="../../denovo/galaxy-spades/">Galaxy-Spades</a>
</li>

        
            
<li >
    <a href="../../denovo/galaxy-velvet/galaxy-velvet/">Galaxy-Velvet</a>
</li>

        
    </ul>
  </li>

        
            
  <li class="dropdown-submenu">
    <a tabindex="-1" href="">Annotation</a>
    <ul class="dropdown-menu">
        
            
<li >
    <a href="../../anno/prokka/">Prokka in Galaxy</a>
</li>

        
            
<li >
    <a href="../../anno/prokka_cmdline/">Prokka on commandline</a>
</li>

        
    </ul>
  </li>

        
            
  <li class="dropdown-submenu">
    <a tabindex="-1" href="">Variant finding</a>
    <ul class="dropdown-menu">
        
            
<li class="active">
    <a href="./">Snippy on commandline</a>
</li>

        
    </ul>
  </li>

        
            
  <li class="dropdown-submenu">
    <a tabindex="-1" href="">Pan genome</a>
    <ul class="dropdown-menu">
        
            
<li >
    <a href="../../pan/roary/">Roary</a>
</li>

        
    </ul>
  </li>

        
    </ul>
  </li>

                    
                    </ul>
                </li>
            
            
            
                <li >
                    <a href="../../../about/">About</a>
                </li>
            
            
            </ul>

            <!-- Search, Navigation and Repo links -->
            <ul class="nav navbar-nav navbar-right">
                <li>
                    <a href="#" data-toggle="modal" data-target="#mkdocs_search_modal">
                        <i class="fa fa-search"></i> Search
                    </a>
                </li>
                <li >
                    <a rel="next" href="../../anno/prokka_cmdline/">
                        <i class="fa fa-arrow-left"></i> Previous
                    </a>
                </li>
                <li >
                    <a rel="prev" href="../../pan/roary/">
                        Next <i class="fa fa-arrow-right"></i>
                    </a>
                </li>
                
            </ul>
        </div>
    </div>
</div>

        <div class="container">
            <div class="col-md-3"><div class="bs-sidebar hidden-print affix well" role="complementary">
    <ul class="nav bs-sidenav">
    
        <li class="main active"><a href="#snippy">Snippy</a></li>
        
            <li><a href="#pre-requisites">Pre-requisites</a></li>
        
            <li><a href="#start">Start</a></li>
        
            <li><a href="#input">Input</a></li>
        
            <li><a href="#how-it-works">How it works</a></li>
        
            <li><a href="#run-snippy">Run Snippy</a></li>
        
            <li><a href="#output">Output</a></li>
        
            <li><a href="#next">Next</a></li>
        
    
    </ul>
</div></div>
            <div class="col-md-9" role="main">

<h1 id="snippy">Snippy</h1>
<p>cmdline</p>
<p>This tutorial will demonstrate how to find variants in a bacterial genome using Snippy. Variants are found by comparing to a reference genome of the same species.</p>
<p><a href="https://github.com/tseemann/snippy/blob/master/README.md#snippy">Github link to Snippy</a></p>
<h2 id="pre-requisites">Pre-requisites</h2>
<ul>
<li>connect to your GVL - cmdline</li>
<li>background knowledge: variant calling</li>
</ul>
<h2 id="start">Start</h2>
<ul>
<li>log in to your virtual machine via terminal</li>
<li>navigate to the place where you want Snippy to run.</li>
<li>make a folder called snippy - <code>mkdir snippy</code></li>
<li>move into that folder - <code>cd snippy</code></li>
</ul>
<h2 id="input">Input</h2>
<h3 id="raw-sequence-reads">Raw sequence reads</h3>
<ul>
<li>Illumina paired-end reads from a bacteria in FASTQ format.</li>
<li>These reads are from <em>Pasteurella multocida</em>, from EMBL-EBI ENA. We will use <code>wget</code> to download them via FTP.</li>
<li>in the snippy folder,</li>
</ul>
<pre><code class="bash">wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR125/003/SRR1257473/SRR1257473_1.fastq.gz
</code></pre>

<pre><code class="bash">wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR125/003/SRR1257473/SRR1257473_2.fastq.gz
</code></pre>

<ul>
<li>These files are compressed and so end in .gz. Snippy can use these .gz read files directly without unzipping.</li>
</ul>
<h3 id="reference-genome">Reference genome</h3>
<ul>
<li>Reference genome from the same species, <em>Pasteurella multocida</em>, from EMBL-EBI Ensembl genomes, in FASTA format.</li>
<li>in the snippy folder,</li>
</ul>
<pre><code class="bash">wget ftp://ftp.ensemblgenomes.org/pub/bacteria/release-31/fasta/bacteria_104_collection/pasteurella_multocida_subsp_multocida_gca_001027695/dna/Pasteurella_multocida_subsp_multocida_gca_001027695.ASM102769v1.31.dna.genome.fa.gz
</code></pre>

<ul>
<li>This file is also compressed into .gz format. This needs to be unzipped:</li>
</ul>
<pre><code class="bash">gunzip ftp://ftp.ensemblgenomes.org/pub/bacteria/release-31/fasta/bacteria_104_collection/pasteurella_multocida_subsp_multocida_gca_001027695/dna/Pasteurella_multocida_subsp_multocida_gca_001027695.ASM102769v1.31.dna.genome.fa.gz
</code></pre>

<ul>
<li>The file will now end in .fa (which is fasta format, and Snippy can use).</li>
</ul>
<h2 id="how-it-works">How it works</h2>
<ul>
<li>Reads are mapped to the reference genome using BWA: this makes a BAM file</li>
<li>BAM file and the ref genome sequence sent to Freebayes</li>
<li>Freebayes finds differences between the reads and the reference, and calls the variants.</li>
</ul>
<h2 id="run-snippy">Run Snippy</h2>
<ul>
<li>cpus: choose number of cpus to use [or it uses a default] - here we will use 16</li>
<li>outdir: choose a name for the output directory, where results will go - here we will use &ldquo;mysnps&rdquo;</li>
<li>ref: the input reference genome filename</li>
<li>R1: the input R1 reads filename</li>
<li>R2: the input R2 reads filename</li>
<li>to run snippy:</li>
</ul>
<pre><code class="bash">snippy --cpus 16 --outdir mysnps --ref [filename.fa] --R1 [R1.fastq.gz] --R2 [R2.fastq.gz]
</code></pre>

<h2 id="output">Output</h2>
<ul>
<li>17 output files</li>
<li>list all the output files (that were put into the &ldquo;mysnps&rdquo; folder):</li>
</ul>
<pre><code class="bash">ls mysnps
</code></pre>

<ul>
<li>look at the first 10 lines of the snps.tab file</li>
</ul>
<pre><code class="bash">head -10 mysnps/snps.tab
</code></pre>

<ul>
<li>look at these columns: chromosome (CHROM), genomic position (POS), variant type (TYPE), nucleotide state in the ref (REF), nucleotide state in the input sample (ALT), and the frequency counts of REF and ALT (EVIDENCE).</li>
<li>FIXME: screenshot with arrows</li>
<li>FIXME: filter for quality?</li>
<li>FIXME: load reference and the tabular vcf file into JBrowse/Artemis/IGV to view the genome and the snps.</li>
<li>FIXME: is there anything we are looking for in particular? e.g. number of variants, existing known variants, variants in particular genes, AMR variants?</li>
</ul>
<h2 id="next">Next</h2></div>
        </div>

        <footer class="col-md-12">
            <hr>
            
            <center>Documentation built with <a href="http://www.mkdocs.org/">MkDocs</a>.</center>
        </footer>

        <script src="../../../js/jquery-1.10.2.min.js"></script>
        <script src="../../../js/bootstrap-3.0.3.min.js"></script>
        <script src="../../../js/highlight.pack.js"></script>
        <script>var base_url = '../../..';</script>
        <script data-main="../../../mkdocs/js/search.js" src="../../../mkdocs/js/require.js"></script>
        <script src="../../../js/base.js"></script>

        <div class="modal" id="mkdocs_search_modal" tabindex="-1" role="dialog" aria-labelledby="Search Modal" aria-hidden="true">
            <div class="modal-dialog">
                <div class="modal-content">
                    <div class="modal-header">
                        <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
                        <h4 class="modal-title" id="exampleModalLabel">Search</h4>
                    </div>
                    <div class="modal-body">
                        <p>
                            From here you can search these documents. Enter
                            your search terms below.
                        </p>
                        <form role="form">
                            <div class="form-group">
                                <input type="text" class="form-control" placeholder="Search..." id="mkdocs-search-query">
                            </div>
                        </form>
                        <div id="mkdocs-search-results"></div>
                    </div>
                    <div class="modal-footer">
                    </div>
                </div>
            </div>
        </div>
    </body>
</html>